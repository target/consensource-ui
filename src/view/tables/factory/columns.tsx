import React from 'react';
import { MUIDataTableColumn, MUIDataTableColumnOptions } from 'mui-datatables';
import {
  FilterTextField,
  FilterMultiselect,
  CertificationsMultiselect,
  CertificatesCell,
  FactoryNameCell,
} from './components';

export const DEFAULT_CELL_VALUE = '-';
/**
 * Used in the `customBodyRender` config. If an empty string is passed,
 * replace it with a placeholder text (`'-'`).
 */
export const getCellValOrDefault = (value: string) =>
  value || DEFAULT_CELL_VALUE;

/**
 * Return the string value for the filter chips displayed
 * in the `<TableFilterList />`.
 *
 * For columns with a single filter value, a single string is passed.
 *
 * For columns with a multiselect (or custom columns), an array of
 * strings is passed.
 *
 */
export const getCustomFilterChipLabel = (
  colLabel: string,
  values: string | string[],
) => {
  const getLabel = (val: string) => `${colLabel}: ${val}`;

  return typeof values === 'string' ? getLabel(values) : values.map(getLabel);
};

/**
 * Configuration for a custom search filter.
 */
export const getCustomSearchOptions = (
  colLabel: string,
): MUIDataTableColumnOptions => ({
  filter: true,
  filterType: 'custom',
  customBodyRender: getCellValOrDefault,
  customFilterListOptions: {
    render: (value) => getCustomFilterChipLabel(colLabel, value),
  },
  filterOptions: {
    display: (filterList, onChange, index, column) => {
      return (
        <FilterTextField
          filterVal={filterList[index]}
          queryKey={column.name}
          label={colLabel}
          onChange={(val) => onChange(val, index, column)}
        />
      );
    },
  },
});

/**
 * Configuration for our custom certificate filtering logic.
 */
export const nameColumnOptions: MUIDataTableColumn['options'] = {
  sort: false,
  filterType: 'custom',
  // TODO: Verify with real cert data below
  customBodyRender: ({ name, id }: any) => (
    <FactoryNameCell name={name} id={id} />
  ),
  customFilterListOptions: {
    render: ({ name }: any) => getCustomFilterChipLabel('Name', name),
  },
  filterOptions: {
    display: (filterList, onChange, index, column) => (
      <FilterTextField
        filterVal={filterList[index]}
        queryKey="name"
        label="Name"
        onChange={(val) => onChange(val, index, column)}
      />
    ),
  },
};

/**
 * Configuration for our custom certificate filtering logic.
 */
export const certColumnOptions: MUIDataTableColumn['options'] = {
  sort: false,
  filterType: 'custom',
  // TODO: Verify with real cert data below
  customBodyRender: (value) => <CertificatesCell certificates={value as any} />,
  customFilterListOptions: {
    render: (value) => getCustomFilterChipLabel('Certification', value),
  },
  filterOptions: {
    display: (filterList, onChange, index, column) => (
      <CertificationsMultiselect
        activeCertFilters={filterList[index]}
        onChange={(val) => onChange(val, index, column)}
      />
    ),
  },
};

/**
 * Configuration for our custom country filtering logic.
 */
export const countryColumnOptions: MUIDataTableColumn['options'] = {
  filterType: 'custom',
  customBodyRender: getCellValOrDefault,
  customFilterListOptions: {
    render: (value) => getCustomFilterChipLabel('Country', value),
  },
  filterOptions: {
    display: (filterList, onChange, index, column, filterData) => (
      <FilterMultiselect
        // TODO: Currently this will only get the countries on the current page.
        options={filterData[index]}
        queryKey="country"
        label="Countries"
        onChange={(val) => onChange(val, index, column)}
        filterVals={filterList[index]}
      />
    ),
  },
};

/**
 * Configuration logic for the keys of our flattened `FactoryResData` object.
 * Other tables can expand upon this with additional columns, such as a link
 * to the factory profile page.
 *
 * The `name` key for each config object corresponds to the key of the object
 * array that is passed to the `data` prop on the table. The `label` key is
 * the string used as the column header.
 *
 * [Details on the various `options` configs can be found here](https://github.com/gregnb/mui-datatables#customize-columns).
 */
export const baseFactoryTableCols: MUIDataTableColumn[] = [
  { name: 'name', label: 'Name', options: nameColumnOptions },
  {
    name: 'certificates',
    label: 'Certifications',
    options: certColumnOptions,
  },
  {
    name: 'country',
    label: 'Country',
    options: countryColumnOptions,
  },
  {
    name: 'street_line_1',
    label: 'Street Line 1',
    options: getCustomSearchOptions('Street Line 1'),
  },
  {
    name: 'street_line_2',
    label: 'Street Line 2',
    options: getCustomSearchOptions('Street Line 2'),
  },
  {
    name: 'city',
    label: 'City',
    options: getCustomSearchOptions('City'),
  },
  {
    name: 'state_province',
    label: 'State/Province',
    options: getCustomSearchOptions('State/Province'),
  },
  {
    name: 'postal_code',
    label: 'Postal Code',
    options: getCustomSearchOptions('Postal Code'),
  },
];
