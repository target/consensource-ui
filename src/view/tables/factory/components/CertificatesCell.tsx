import React from 'react';
import { CertResData } from 'services/api';
import {
  List,
  ListItem,
  Tooltip,
  makeStyles,
  createStyles,
} from '@material-ui/core';
import { UnstyledLink } from 'view/components';

export interface CertificatesCellProps {
  // TODO: Remove optional `certificates` once we fix backend issues
  certificates?: CertResData[];
}

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    link: {
      textDecoration: 'none',
      color: 'inherit',
    },
    certName: {
      borderBottom: `2px solid ${palette.primary.light}`,
      cursor: 'pointer',
      '&:hover': {
        borderBottom: `2px solid ${palette.primary.main}`,
      },
    },
  }),
);

export const CertificatesCell = ({ certificates }: CertificatesCellProps) => {
  const classes = useStyles();

  if (!certificates || certificates.length === 0) {
    return (
      <ListItem>
        <i>None</i>
      </ListItem>
    );
  }

  return (
    <List>
      {certificates.map(({ standard_name, id, valid_to }) => (
        <UnstyledLink to={`certifications/${id}`}>
          <ListItem key={id}>
            <Tooltip
              placement="top"
              title={`Valid until ${new Date(valid_to).toLocaleDateString()}`}
            >
              <i className={classes.certName}>{standard_name}</i>
            </Tooltip>
          </ListItem>
        </UnstyledLink>
      ))}
    </List>
  );
};
