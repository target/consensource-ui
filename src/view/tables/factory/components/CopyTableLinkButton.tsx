import React from 'react';
import { useStores } from 'services/hooks';
import { AttachFile as PaperClipIcon } from '@material-ui/icons';
import CopyToClipboard, {
  Props as CopyToClipboardProps,
} from 'react-copy-to-clipboard';
import {
  Tooltip,
  IconButton,
  createStyles,
  makeStyles,
} from '@material-ui/core';

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    icon: {
      '&:hover': {
        color: palette.primary.main,
      },
    },
  }),
);

export const CopyTableLinkButton = () => {
  const classes = useStyles();
  const { snackbarStore } = useStores();

  const handleClick: CopyToClipboardProps['onCopy'] = (text, result) => {
    if (result) {
      snackbarStore.open('Copied link');
    }
  };

  return (
    <CopyToClipboard text={window.location.href} onCopy={handleClick}>
      <Tooltip title="Copy Link To Table">
        <IconButton className={classes.icon}>
          <PaperClipIcon />
        </IconButton>
      </Tooltip>
    </CopyToClipboard>
  );
};
