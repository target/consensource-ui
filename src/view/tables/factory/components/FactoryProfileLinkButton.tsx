import React from 'react';
import { Link } from 'react-router-dom';
import { OrgResData } from 'services/api';
import { OpenInNew as OpenInNewIcon } from '@material-ui/icons';
import {
  Tooltip,
  IconButton,
  createStyles,
  makeStyles,
} from '@material-ui/core';

export interface FactoryProfileLinkButtonProps {
  factoryId: OrgResData['id'];
}

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    icon: {
      '&:hover': {
        color: palette.primary.main,
      },
    },
  }),
);

export const FactoryProfileLinkButton = ({
  factoryId,
}: FactoryProfileLinkButtonProps) => {
  const classes = useStyles();

  return (
    <Link to={{ pathname: `factories/${factoryId}` }}>
      <Tooltip title="View Factory Profile">
        <IconButton className={classes.icon}>
          <OpenInNewIcon />
        </IconButton>
      </Tooltip>
    </Link>
  );
};
