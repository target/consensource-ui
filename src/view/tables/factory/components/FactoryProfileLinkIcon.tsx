import React, { useState } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { SvgIconProps, IconButton, Tooltip } from '@material-ui/core';
import { OpenInNew as OpenInNewIcon } from '@material-ui/icons';
import { OrgResData } from 'services/api';

export interface FactoryProfileLinkIconProps {
  factoryId: OrgResData['id'];
}

export function FactoryProfileLinkIcon({
  factoryId,
}: FactoryProfileLinkIconProps) {
  const { pathname } = useLocation();
  const [color, setColor] = useState<SvgIconProps['color']>('inherit');

  return (
    <Link
      to={{ pathname: `factories/${factoryId}`, state: { from: pathname } }}
    >
      <Tooltip title="View factory profile">
        <IconButton
          onMouseOver={() => setColor('primary')}
          onFocus={() => setColor('primary')}
          onMouseOut={() => setColor('inherit')}
          onBlur={() => setColor('inherit')}
          aria-label="View factory profile"
        >
          <OpenInNewIcon color={color} />
        </IconButton>
      </Tooltip>
    </Link>
  );
}
