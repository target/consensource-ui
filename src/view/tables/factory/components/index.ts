export * from './CertificatesCell';
export * from './FactoryProfileLinkIcon';
export * from './TableTitle';
export * from './filtering';
