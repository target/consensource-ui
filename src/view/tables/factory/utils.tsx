import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
