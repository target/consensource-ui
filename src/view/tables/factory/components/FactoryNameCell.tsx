import React from 'react';
import { FactoryResData } from 'services/api';
import { Tooltip } from '@material-ui/core';
import { Link } from 'view/components';

export interface FactoryNameCellProps {
  name: FactoryResData['name'];
  id: FactoryResData['id'];
}

export const FactoryNameCell = ({ name, id }: FactoryNameCellProps) => {
  return (
    <Link to={`factories/${id}`} color="secondary">
      <Tooltip placement="top" title="View factory profile">
        <i>{name}</i>
      </Tooltip>
    </Link>
  );
};
