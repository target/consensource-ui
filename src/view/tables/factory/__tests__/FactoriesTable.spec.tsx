import React from 'react';
import { mockFactoryResWithCerts } from 'services/api/__mocks__';
import { render, screen, act } from 'utils/testing';
import userEvent from '@testing-library/user-event';
import { FactoryResData, PaginatedApiRes } from 'services/api';
import { FactoriesTable, textLabels } from '..';
import { columns, DEFAULT_CELL_VALUE } from '../columns';
import { FILTER_TIMEOUT_MS } from '../components';

// Mocked since it makes an api call to populate the multiselect
jest.mock('../components/filtering/CertificationsMultiselect', () => {
  return {
    CertificationsMultiselect: () => <div />,
  };
});

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<FactoriesTable />', () => {
  const mockFactories: PaginatedApiRes<
    FactoryResData[]
  > = mockFactoryResWithCerts as any;

  describe('general', () => {
    it('renders an empty table', () => {
      render(
        <FactoriesTable
          factories={{ ...mockFactories, data: [] }}
          searchParams={{}}
        />,
      );

      expect(
        screen.getByText(textLabels!.body!.noMatch! as any),
      ).toBeInTheDocument();
    });

    it('renders a table with data', async () => {
      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);
      expect(screen.getByTestId('MUIDataTableBodyRow-0')).toBeInTheDocument();
    });

    it('renders a default text value for empty fields', async () => {
      render(
        <FactoriesTable
          factories={{ ...mockFactories, data: [{}] } as any}
          searchParams={{}}
        />,
      );

      const firstRow = screen.getByTestId('MuiDataTableBodyCell-4-0');
      const dataCell = firstRow.childNodes[1];

      expect(dataCell.textContent).toBe(DEFAULT_CELL_VALUE);
    });

    it('sets pagination controls and total factory count', () => {
      const limit = 10;
      const offset = 20;
      const total = 40;

      // Search params are parsed as strings in the component
      const searchParams = {
        offset: offset.toString(),
        limit: limit.toString(),
      };

      render(
        <FactoriesTable
          factories={{ ...mockFactories, paging: { total } } as any}
          searchParams={searchParams}
        />,
      );

      const paginationControl = screen.getByTestId('pagination-rows');

      expect(paginationControl.firstChild?.nodeValue).toEqual(limit.toString());
      expect(
        screen.getByText(`${offset + 1}-${offset + limit} of ${total}`),
      ).toBeInTheDocument();
    });
  });

  describe('search bar', () => {
    it('renders as closed by default', () => {
      expect(screen.queryByTitle('Search')).toBe(null);
    });

    it('renders placeholder text', () => {
      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);

      userEvent.click(screen.getByTitle('Search'));
      expect(
        screen.getByPlaceholderText('Search by name, certifications...'),
      ).toBeInTheDocument();
    });

    it('sets the search text to the search params address string', () => {
      const address = 'address';

      render(
        <FactoriesTable factories={mockFactories} searchParams={{ address }} />,
      );

      userEvent.click(screen.getByTitle('Search'));
      expect(screen.getByDisplayValue(address)).toBeInTheDocument();
    });

    it('sets the search text to an empty string if search params address is not a string', async () => {
      const address = ['foo', 'bar'];

      render(
        <FactoriesTable factories={mockFactories} searchParams={{ address }} />,
      );

      userEvent.click(screen.getByTitle('Search'));
      expect(screen.getByDisplayValue('')).toBeInTheDocument();
    });
  });

  describe('filtering', () => {
    const { name: validFilterName, label: validFilterLabel } = columns[5];
    const filterVal = 'foo';

    it('renders', () => {
      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);

      userEvent.click(screen.getByTitle('Filter Table'));
      const container = screen.getByRole('presentation');

      expect(container).toMatchSnapshot();
    });

    // The only filters that we display as a filter chip are those
    // that have a display column. For example, `limit` is a valid
    // query param, but we don't display a filter chip since it is
    // displayed in the table footer.
    it('renders a filter chip from valid search params', () => {
      render(
        <FactoriesTable
          factories={mockFactories}
          searchParams={{ [validFilterName]: filterVal }}
        />,
      );

      expect(
        screen.getByText(`${validFilterLabel}: ${filterVal}`),
      ).toBeInTheDocument();
    });

    it('does not render a filter chip from invalid search params', () => {
      const invalidFilterName = 'invalid_fitestlter';

      render(
        <FactoriesTable
          factories={mockFactories}
          searchParams={{ [invalidFilterName]: filterVal }}
        />,
      );

      expect(screen.queryByText(`${invalidFilterName}: ${filterVal}`)).toBe(
        null,
      );
    });

    it('renders multiple filter chips for multiple filter values', () => {
      const filterVal1 = 'foo';
      const filterVal2 = 'bar';

      render(
        <FactoriesTable
          factories={mockFactories}
          searchParams={{
            [validFilterName]: [filterVal1, filterVal2],
          }}
        />,
      );

      expect(
        screen.getByText(`${validFilterLabel}: ${filterVal1}`),
      ).toBeInTheDocument();
      expect(
        screen.getByText(`${validFilterLabel}: ${filterVal2}`),
      ).toBeInTheDocument();
    });

    it('updates the query string when a filter chip is removed', () => {
      render(
        <FactoriesTable
          factories={mockFactories}
          searchParams={{ [validFilterName]: filterVal }}
        />,
      );

      const filterChip = screen.getByText(`${validFilterLabel}: ${filterVal}`);
      userEvent.click(filterChip.nextSibling as any);

      expect(mockHistoryPush).toHaveBeenCalledWith({
        search: '',
      });
    });

    it('updates the query string when a new filter is applied', () => {
      jest.useFakeTimers();

      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);

      userEvent.click(screen.getByTitle('Filter Table'));
      userEvent.type(screen.getByLabelText(validFilterLabel as any), filterVal);
      userEvent.click(screen.getByTitle('apply filters'));

      act(() => {
        jest.advanceTimersByTime(FILTER_TIMEOUT_MS);
      });

      expect(mockHistoryPush).toHaveBeenCalledWith({
        search: `${validFilterName}=${filterVal}`,
      });
    });
  });

  describe('sorting', () => {
    const { name: sortColName } = columns[2];

    it('when single clicking a column header, sets the query string to ascending for the column', () => {
      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);
      userEvent.click(screen.getByTestId('headcol-2'));

      expect(mockHistoryPush).toHaveBeenCalledWith({
        search: `sort_dir=asc&sort_key=${sortColName}`,
      });
    });

    it('when double clicking a column header, sets the query string to descending for the column', () => {
      render(<FactoriesTable factories={mockFactories} searchParams={{}} />);
      userEvent.click(screen.getByTestId('headcol-2'));
      userEvent.click(screen.getByTestId('headcol-2'));

      expect(mockHistoryPush).toHaveBeenCalledWith({
        search: `sort_dir=desc&sort_key=${sortColName}`,
      });
    });
  });
});
