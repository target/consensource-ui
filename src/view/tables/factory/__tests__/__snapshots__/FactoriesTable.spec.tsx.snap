// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FactoriesTable /> filtering renders 1`] = `
<div
  class="MuiPopover-root"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiPaper-root MuiPopover-paper MUIDataTableToolbar-filterPaper MuiPaper-elevation2 MuiPaper-rounded"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
    tabindex="-1"
  >
    <button
      aria-label="Close"
      class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-filterCloseIcon"
      style="position: absolute; right: 4px; top: 4px; z-index: 1000;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MUIDataTableFilter-root"
    >
      <div
        class="MUIDataTableFilter-header"
      >
        <div
          class="MUIDataTableFilter-reset"
        >
          <p
            class="MuiTypography-root MUIDataTableFilter-title MuiTypography-body2"
          >
            FILTERS
          </p>
          <button
            aria-label="RESET"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableFilter-resetLink MuiButton-textPrimary"
            data-testid="filterReset-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              RESET
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MUIDataTableFilter-filtersSelected"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="name"
                id="name-label"
              >
                Name
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Countries
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-Countries"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-Countries"
                role="button"
                tabindex="0"
              >
                <span>
                  â€‹
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="Countries"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="street-line-1"
                id="street-line-1-label"
              >
                Street Line 1
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="street-line-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="street-line-2"
                id="street-line-2-label"
              >
                Street Line 2
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="street-line-2"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="city"
                id="city-label"
              >
                City
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="city"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="state/province"
                id="state/province-label"
              >
                State/Province
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="state/province"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MUIDataTableFilter-gridListTile"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                for="postal-code"
                id="postal-code-label"
              >
                Postal Code
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="postal-code"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-top: 40px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          title="apply filters"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Apply Filters
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
